velox/vector/ComplexVector.cpp

```diff
@@ -1781,4 +1782,266 @@ void RowVector::appendNulls(vector_size_t numberOfRows) {
   bits::fillBits(mutableRawNulls(), oldSize, newSize, bits::kNull);
 }
 
+namespace {
+template <typename KeyType, TypeKind ValueKind>
+std::vector<VectorPtr> toRowVectorComplexImpl(
+    const BaseVector& vector,
+    folly::F14FastMap<KeyType, size_t> keyToIndex,
```

Comment:
nit: perhaps, add a @param keyToIndex to clarify that 'index' is unique and comes from [0, keyToIndex.size()) range.